src/compiler/checker.ts

```diff
@@ -6532,9 +6530,14 @@ module ts {
         }
 
         function getTypeAtPosition(signature: Signature, pos: number): Type {

```

Comment:
I would far prefer to have two separate functions, one for spread, and one for normal arguments. I noticed above that you use the conditional operator to switch the pos value you pass in, so instead you can just switch the function that you call, and pass the pos as is.